libsel4muslcsys/src/sys_io.c

```diff
@@ -253,8 +253,6 @@ long sys_close(va_list ap)
 
     if (fds->filetype == FILE_TYPE_CPIO) {
         free(fds->data);
-    } else {
```

Comment:
maybe it's better to check for all known types explicitly and leave the unconditional assert in the end.
```
} else if (fds->filetype == FILE_TYPE_SOCKET) {
    /* nothing to do */
} else {
    assert(!"not implemented");
}
```

Reply:
This assert is here because this syscall only knows how to handle FILE_TYPE_CPIO file types. libsel4camkes provides an extension to this syscall that knows how to close the FILE_TYPE_SOCKET that camkes uses:
https://github.com/seL4/camkes-tool/blob/master/libsel4camkes/src/sys_io.c#L47
These syscalls can be installed by calling `camkes_install_io_syscalls()`